docs/user-guide/federation/namespaces.md

```diff
@@ -0,0 +1,90 @@
+---
+---
+
+This guide explains how to use namespaces in Federation control plane.
+
+
+* TOC
+{:toc}
+
+## Prerequisites
+
+This guide assumes that you have a running Kubernetes Cluster
+Federation installation. If not, then head over to the
+[federation admin guide](/docs/admin/federation/) to learn how to
+bring up a cluster federation (or have your cluster administrator do
+this for you). Other tutorials, for example
+[this one](https://github.com/kelseyhightower/kubernetes-cluster-federation)
+by Kelsey Hightower, are also available to help you.
+
+You are also expected to have a basic
+[working knowledge of Kubernetes](/docs/getting-started-guides/) in
+general and [Namespaces](/docs/user-guide/namespaces/) in particular.
+
+## Overview
+
+Namespaces in federation control plane (referred to as "federated namespaces" in
+this guide) are very similar to the traditional [Kubernetes
+Namespaces](/docs/user-guide/namespaces/) providing the same functionality.
+Creating them in the federation control plane ensures that they are synchronized
+across all the clusters in federation.
+
+
+## Creating a Federated Namespace
+
+The API for Federated Namespaces is 100% compatible with the
+API for traditional Kubernetes Namespaces. You can create a namespace by sending
+a request to the federation apiserver.
+
+You can do that using kubectl by running:
+
+``` shell
+kubectl --context=federation-cluster create -f myns.yaml
```

Comment:
Yes myns.yaml is just an example file name. Any yaml describing a namespace is fine.
That failed to ensure finalizer error is fine as long as it eventually succeeds (federated namespace controller keeps retrying).

Namespace remains in terminating state until all resources in that namespace are deleted. Can you list all the resources in the namespace?

Please feel free to file a separate issue. `kubectl delete ns` should work. Its a bug if it doesnt.